src/components/tl/cuda/bcast/bcast_linear.c

```diff
@@ -0,0 +1,303 @@
+/**
+ * Copyright (c) 2024, NVIDIA CORPORATION & AFFILIATES. All rights reserved.
+ *
+ * See file LICENSE for terms.
+ */
+
+#include "bcast/bcast.h"
+
+enum
+{
+    STAGE_DONE,
+    STAGE_SYNC,
+    STAGE_SETUP,
+    // root
+    STAGE_COPY,      // post copy task: copy block from src to scratch buffer
+    STAGE_WAIT_COPY, // wait for copy finishes
+    STAGE_WAIT_ALL,  // wait for all others rank be on same step
+    // non-root
+    STAGE_WAIT_ROOT,   // clients wait while root writes to own scratch buffer
+    STAGE_CLIENT_COPY, // clients submit copy task
+    STAGE_CLIENT_COPY_WAIT, // clients wait completion of copy from root's scratch
+};
+
+ucc_status_t ucc_tl_cuda_bcast_linear_setup_start(ucc_tl_cuda_task_t *task)
+{
+    ucc_tl_cuda_team_t *team  = TASK_TEAM(task);
+    ucc_rank_t          trank = UCC_TL_TEAM_RANK(team);
+    ucc_status_t        status;
+
+    set_rank_step(task, trank, 0, 0);
+    ucc_memory_cpu_store_fence();
+    status = ucc_tl_cuda_shm_barrier_start(UCC_TL_TEAM_RANK(team), task->bar);
+    if (ucc_unlikely(status != UCC_OK)) {
+        goto exit_err;
+    }
+
+    return UCC_OK;
+
+exit_err:
```

Comment:
```suggestion
```
I understand it comes from the codebase though

Reply:
removed lines and check format again

Reply:
what about that?

Reply:
done